huaweicloud/services/dc/resource_huaweicloud_dc_hosted_connect.go

```diff
@@ -0,0 +1,446 @@
+				Description: `The description of the hosted connect.`,
+			},
+			"peer_location": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `The location of the on-premises facility at the other end of the connection.`,
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The status of the hosted connect.`,
+			},
+		},
+	}
+}
+
+func resourceHostedConnectCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var (
+		createHostedConnectHttpUrl = "v3/{project_id}/dcaas/hosted-connects"
+		createHostedConnectProduct = "dc"
+	)
+	createHostedConnectClient, err := cfg.NewServiceClient(createHostedConnectProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DC client: %s", err)
+	}
+
+	createHostedConnectPath := createHostedConnectClient.Endpoint + createHostedConnectHttpUrl
+	createHostedConnectPath = strings.ReplaceAll(createHostedConnectPath, "{project_id}", createHostedConnectClient.ProjectID)
+
+	createHostedConnectOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			201,
+		},
+		MoreHeaders: map[string]string{"Content-Type": "application/json"},
+	}
+
+	createHostedConnectOpt.JSONBody = utils.RemoveNil(buildCreateHostedConnectBodyParams(d))
+	createHostedConnectResp, err := createHostedConnectClient.Request("POST", createHostedConnectPath, &createHostedConnectOpt)
+	if err != nil {
+		return diag.Errorf("error creating hosted connect: %s", err)
+	}
+
+	createHostedConnectRespBody, err := utils.FlattenResponse(createHostedConnectResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, err := jmespath.Search("hosted_connect.id", createHostedConnectRespBody)
+	if err != nil {
+		return diag.Errorf("error creating hosted connect: ID is not found in API response")
+	}
+	d.SetId(id.(string))
+
+	err = hostedConnectWaitingForStateCompleted(ctx, d, meta, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.Errorf("error waiting for the hosted connect (%s) creation to complete: %s", d.Id(), err)
+	}
+	return resourceHostedConnectRead(ctx, d, meta)
+}
+
+func buildCreateHostedConnectBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"hosted_connect": map[string]interface{}{
+			"name":               utils.ValueIngoreEmpty(d.Get("name")),
+			"description":        utils.ValueIngoreEmpty(d.Get("description")),
+			"bandwidth":          d.Get("bandwidth"),
+			"hosting_id":         d.Get("hosting_id"),
+			"vlan":               d.Get("vlan"),
+			"resource_tenant_id": d.Get("resource_tenant_id"),
+			"peer_location":      utils.ValueIngoreEmpty(d.Get("peer_location")),
+		},
+	}
+	return bodyParams
+}
+
+func hostedConnectWaitingForStateCompleted(ctx context.Context, d *schema.ResourceData, meta interface{}, t time.Duration) error {
```

Comment:
can we reuse the client then it's unnecessary to build the client in **Refresh** func.
```suggestion
func hostedConnectWaitingForStateCompleted(ctx context.Context, client *golangsdk.ServiceClient, id string, t time.Duration) error {
```

Reply:
ok，done